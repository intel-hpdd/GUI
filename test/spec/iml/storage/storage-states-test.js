describe('storage states', () => {
  let storageState, addStorageState, storageDetailState;

  beforeEach(() => {
    jest.mock('../../../../source/iml/storage/storage-resolves.js', () => ({
      storageB: 'storageB',
      alertIndicatorB: 'alertIndicatorB',
      getData: 'getData',
      storageResource$: 'storageResource$'
    }));

    ({
      storageState,
      addStorageState,
      storageDetailState
    } = require('../../../../source/iml/storage/storage-states.js'));
  });

  it('should create the storage state', () => {
    expect(addStorageState).toMatchSnapshot();
  });

  it('should create the storage state', () => {
    expect(storageState).toMatchSnapshot();
  });

  it('should create the storage detail state', () => {
    expect(storageDetailState).toMatchSnapshot();
  });
});
