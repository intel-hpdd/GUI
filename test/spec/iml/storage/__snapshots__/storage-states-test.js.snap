// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storage states should create the storage detail state 1`] = `
Object {
  "component": "storageDetail",
  "data": Object {
    "access": "filesystem_administrators",
    "helpPage": "storage_tab.htm",
    "icon": "fa-hdd-o",
    "kind": "Storage Detail",
  },
  "name": "app.storageDetail",
  "params": Object {
    "resetState": Object {
      "dynamic": true,
    },
  },
  "resolve": Object {
    "alertIndicatorB": "alertIndicatorB",
    "getData": "getData",
    "storageResource$": "storageResource$",
  },
  "url": "/configure/storage/:id",
}
`;

exports[`storage states should create the storage state 1`] = `
Object {
  "component": "addStorage",
  "data": Object {
    "access": "filesystem_administrators",
    "helpPage": "storage_tab.htm",
    "icon": "fa-hdd-o",
    "kind": "Add Storage Device",
  },
  "name": "app.addStorage",
  "params": Object {
    "resetState": Object {
      "dynamic": true,
    },
  },
  "resolve": Object {
    "storageB": "storageB",
  },
  "url": "/configure/storage/add",
}
`;

exports[`storage states should create the storage state 2`] = `
Object {
  "component": "storage",
  "data": Object {
    "access": "filesystem_administrators",
    "helpPage": "storage_tab.htm",
    "icon": "fa-hdd-o",
    "kind": "Storage",
  },
  "name": "app.storage",
  "params": Object {
    "resetState": Object {
      "dynamic": true,
    },
  },
  "resolve": Object {
    "alertIndicatorB": "alertIndicatorB",
    "storageB": "storageB",
  },
  "url": "/configure/storage",
}
`;
